test/example/match.cpp

```diff
@@ -362,27 +362,54 @@ void test_internal_binary() {
   std::cout << "Testing Internal::BinaryMatcher\n";
 
   Binary* out = nullptr;
+  BinaryOp op;
 
-  auto eq32Matcher =
+  auto binMatcher =
     Internal::BinaryMatcher(&out,
-                            EqInt32,
+                            Internal::Any<BinaryOp>(&op),
                             Internal::Any<Expression*>(nullptr),
                             Internal::Any<Expression*>(nullptr));
+  assert(binMatcher.matches(eq32));
+  assert(out == eq32);
+  assert(op == EqInt32);
+  assert(binMatcher.matches(eq64));
+  assert(out == eq64);
+  assert(op == EqInt64);
+  assert(binMatcher.matches(add));
+  assert(out == add);
+  assert(op == AddInt32);
+  assert(!binMatcher.matches(nop));
+
+  assert(matches(eq32, binary(any(), any())));
+  assert(matches(eq64, binary(&out, any(), any())));
```

Comment:
Could it be also defined as:
```cpp
matches(eq32, binary(&out));
```
which will be equivalent to
```cpp
matches(eq32, binary(&out, any(), any()));
```

Reply:
Sure, I can add that overload and do the same for unary.

Reply:
Actually, thinking about this more, I think we should always pass submatchers to `binary`, even if they are both just `any()`. Otherwise there will just be too many overloads of `binary` that don't really add new functionality. I'll still do this for `unary`, though.